druid/src/main/java/org/apache/calcite/adapter/druid/DruidQuery.java

```diff
@@ -833,6 +863,54 @@ protected JsonAggregation getJsonAggregation(List<String> fieldNames,
     }
   }
 
+  public JsonPostAggregation getJsonPostAggregation(String name, RexNode rexNode, RelNode rel) {
+    if (rexNode instanceof RexCall) {
+      List<JsonPostAggregation> fields = new ArrayList<>();
+      for (RexNode ele : ((RexCall) rexNode).getOperands()) {
+        JsonPostAggregation field = getJsonPostAggregation("", ele, rel);
+        if (field == null) {
+          return null;
+        }
+        fields.add(field);
+      }
+      switch (rexNode.getKind()) {
+      case PLUS:
+        return new JsonArithmetic(name, "+", fields, null);
+      case MINUS:
+        return new JsonArithmetic(name, "-", fields, null);
+      case DIVIDE:
+        return new JsonArithmetic(name, "quotient", fields, null);
+      case TIMES:
+        return new JsonArithmetic(name, "*", fields, null);
+      case CAST:
+        return getJsonPostAggregation(name, ((RexCall) rexNode).getOperands().get(0),
+            rel);
+      default:
+      }
+    } else if (rexNode instanceof RexInputRef) {
+      if (rel instanceof Aggregate) {
+        Integer indexSkipGroup = ((RexInputRef) rexNode).getIndex()
+            - ((Aggregate) rel).getGroupSet().cardinality();
+        AggregateCall aggCall = ((Aggregate) rel).getAggCallList().get(indexSkipGroup);
+        if (aggCall.isDistinct() && aggCall.getAggregation().getKind().equals(SqlKind.COUNT)) {
+          // Will be a hyper unique cardinality column.
+          // Use hyperUniqueCardinality post aggregator instead of field Accessor.
+          // TODO: Expect to change after CALC-1787
+          return new JsonHyperUniqueCardinality("",
+              rel.getRowType().getFieldNames().get(((RexInputRef) rexNode).getIndex()));
+        }
+      }
+      return new JsonFieldAccessor("",
+          rel.getRowType().getFieldNames().get(((RexInputRef) rexNode).getIndex()));
+    } else if (rexNode instanceof RexLiteral) {
+      if (((RexLiteral) rexNode).getValue3() instanceof BigDecimal) {
```

Comment:
Constant Post aggregator only support number type constant, according to the [Documentation of RexLiteral](https://github.com/apache/calcite/blob/419b810fe67539cf7a3d653e918352fbaf10e8b0/core/src/main/java/org/apache/calcite/rex/RexLiteral.java#L171-L177), the numeric value of RexLiteral could be only BigDecimal type, so I only allow BigDecimal value in RexLiteral to be pushed in. Will add comment to clarify.